programs/ssl/ssl_test_lib.c

```diff
@@ -535,36 +598,18 @@ int parse_curves(const char *curves, uint16_t *group_list, size_t group_list_len
                 *p++ = '\0';
             }
 
-#if defined(MBEDTLS_ECP_LIGHT)
-            if ((curve_cur = mbedtls_ecp_curve_info_from_name(q)) != NULL) {
-                group_list[i++] = curve_cur->tls_id;
-            } else
-#endif
-#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_EPHEMERAL_ENABLED) && \
-            defined(PSA_WANT_ALG_FFDH)
-            if ((ffdh_group = mbedtls_ssl_ffdh_group_from_name(q)) != 0) {
-                group_list[i++] = ffdh_group;
-            } else
-#endif
-            {
+            if ((curve_tls_id = mbedtls_ssl_get_curve_tls_id_from_name(q)) != 0) {
+                group_list[i++] = curve_tls_id;
+            } else {
                 mbedtls_printf("unknown curve %s\n", q);
-#if defined(MBEDTLS_ECP_LIGHT)
                 mbedtls_printf("supported curves: ");
-                for (curve_cur = mbedtls_ecp_curve_list();
-                     curve_cur->grp_id != MBEDTLS_ECP_DP_NONE;
-                     curve_cur++) {
-                    mbedtls_printf("%s ", curve_cur->name);
-                }
-#endif
-#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_EPHEMERAL_ENABLED) && \
-                defined(PSA_WANT_ALG_FFDH)
-                const uint16_t *supported_ffdh_group = mbedtls_ssl_ffdh_supported_groups();
-                while (*supported_ffdh_group != 0) {
-                    mbedtls_printf("%s ",
-                                   mbedtls_ssl_ffdh_name_from_group(*supported_ffdh_group));
-                    supported_ffdh_group++;
+                const char **supported_curves = mbedtls_ssl_get_supported_curves_list();
```

Comment:
That's a minor remark for a program that's not really used, but instead of making a temporary partial copy we could just go through the original list once and print curves on the fly.

Reply:
Do you mean that `mbedtls_ssl_get_supported_curves_list()` should directly print the supported curves instead of returning the array of pointers (which is more complex and probably not needed)?

Reply:
Yes, it's a newly added function used only in this place, it can just do the printing instead.